msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs

```diff
@@ -337,6 +338,43 @@ void BuildEntitlementFlags (GccOptions gcc)
 			}
 		}
 
+		static string Unquote (string text, int startIndex)
+		{
+			var builder = new StringBuilder ();
```

Comment:
minor, but you should delay creating the `StringBuilder` instance after the first two `return` statements.